library/Fission/DNS.hs

```diff
@@ -0,0 +1,34 @@
+module Fission.DNS (splitRecord) where
+
+import           Fission.Prelude
+
+import           RIO.NonEmpty
+import qualified RIO.Text as Text
+import qualified RIO.List as List
+
+
+-- if less than 256 bytes, return normal. 
+-- if larger then split into chunks & prefix with a 3-char decimal prefix  & semicolon delimiter ("001;")
+splitRecord :: Text -> NonEmpty Text
+splitRecord txt = 
+  if Text.length txt <= 256
+    then 
+      pure txt
+
+    else 
+      Text.chunksOf 252 txt
+      |> List.zip [0..]
+      |> (joinRecord <$>)
+      |> nonEmpty
+      |> maybe (pure "") identity
```

Comment:
Two quick points on style:

1. Pipes _must_ be indented. The above wold not work in a `do` block
2. When piping, start with a bae argument (in this case, `txt |> Text.chunksOf`
3. I find the piped version here much harder to read
4. Why not just keep it consistent with a prefix always?
5. It can’t hurt to break these steps out with names

```haskell 
splitRecord txt = 
  fromMaybe (pure “”) records
  where
    records = joinRecord <$> indexed
    indexed = List.zip [0..] Text.chunksOf 252 txt
```

Reply:
Nice! tbh I kinda forgot about `where` clauses :sweat_smile:  this reads much clearer

Reply:
> Pipes must be indented. The above wold not work in a do block 

:+1:

Reply:
> Why not just keep it consistent with a prefix always?
:thinking: we _can_ but TXT records aren't normally treated that way. so I thought this was actually more straightforward: If you get one record back, that's the record, no parsing needed (which is what you expect from DNS). If you get back multiple records where order matters, then prefixes help order them